src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCharClass.cs

```diff
@@ -446,69 +451,16 @@ public void AddLowercase(CultureInfo culture)
         /// For a single range that's in the set, adds any additional ranges
         /// necessary to ensure that lowercase equivalents are also included.
         /// </summary>
-        private void AddLowercaseRange(char chMin, char chMax)
+        private void AddCaseEquivalenceRange(char chMin, char chMax, CultureInfo culture)
         {
-            int i = 0;
-
-            for (int iMax = s_lcTable.Length; i < iMax;)
-            {
-                int iMid = (i + iMax) >> 1;
-                if (s_lcTable[iMid].ChMax < chMin)
-                {
-                    i = iMid + 1;
-                }
-                else
-                {
-                    iMax = iMid;
-                }
-            }
-
-            if (i >= s_lcTable.Length)
-            {
-                return;
-            }
-
-            char chMinT, chMaxT;
-            LowerCaseMapping lc;
-
-            for (; i < s_lcTable.Length && (lc = s_lcTable[i]).ChMin <= chMax; i++)
+            for (int i = chMin; i <= chMax; i++)
```

Comment:
> for (int i = chMin; i <= chMax; i++) [](http://example.com/codeflow?start=11&length=37)

Is it possible this will add the same equivalences more than once? 

For example, if you have  range like `A`..`z` which encounter `i`, and `I`. Is this will add all `i` equivalence twice?

Reply:
It is possible that this happens but RegexCharClass already handles this nicely today, since even today you can have patterns like `[A-DB-C]` which require us to handle adding a range/character more than once. In fact, RegexCharClass is intelligent enough in such cases to be able to reduce the pattern I mentioned to `[A-D]`.